.test-infra/pipelines/src/main/java/org/apache/beam/testinfra/pipelines/dataflow/DataflowClientFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.testinfra.pipelines.dataflow;
```

Comment:
This is basically a new IO. Is it possible to break it out into a separate pr?

also, it might be better served to live in the IO module?

Reply:
Per the [applyJavaNature `publish=false`](https://github.com/damondouglas/beam/blob/dataflow-api-to-bq-pipeline/.test-infra/pipelines/build.gradle#L27), the entire codebase at .test-infra/pipelines is not meant for use outside this module.

Reply:
I added the `@Internal` annotation to all the classes. Thank you for reviewing and discovering this.

Reply:
That works for me. I'll review this code itself now.